contextily/plotting.py

```diff
@@ -124,25 +125,57 @@ def add_basemap(
     # If local source
     else:
         import rasterio as rio
-
         # Read file
         with rio.open(url) as raster:
-            image = np.array([band for band in raster.read()])
+            if reset_extent:
+                from rasterio.mask import mask as riomask
+                # Read window
+                if crs:
+                    left, bottom, right, top = rio.warp.transform_bounds(crs,
+                                                                         raster.crs,
+                                                                         xmin,
+                                                                         ymin,
+                                                                         xmax,
+                                                                         ymax
+                                                                         )
+                else:
+                    left, bottom, right, top = xmin, ymin, xmax, ymax
+                window = [{'type': 'Polygon',
+                          'coordinates': (((left, bottom),
+                                           (right, bottom),
+                                           (right, top),
+                                           (left, top),
+                                           (left, bottom)),)}]
+                image, img_transform = riomask(raster, window, crop=True)
+            else:
+                # Read full
+                image = np.array([band for band in raster.read()])
+                img_transform = raster.transform
             # Warp
             if (crs is not None) and (raster.crs != crs):
                 image, raster = _warper(
-                    image, raster.transform, raster.crs, crs, resampling
+                    image, img_transform, raster.crs, crs, resampling
                 )
             image = image.transpose(1, 2, 0)
             bb = raster.bounds
             extent = bb.left, bb.right, bb.bottom, bb.top
     # Plotting
+    if image.shape[2] == 1:
+        image = image[:, :, 0]
+        print(image.shape)
+        print(extent)
```

Comment:
`print` calls should probably should be removed

Reply:
whoops! I thought that was removed... will do